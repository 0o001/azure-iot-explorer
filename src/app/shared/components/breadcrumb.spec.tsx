/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Breadcrumb from './breadcrumb';

describe('components/shared/breadcrumb', () => {
    const testWithLocalizationContext = (Target: JSX.Element) => {
        const outerWrapper = shallow(Target);
        const Children = outerWrapper.props().children;
        const t = jest.fn((value: string, options: unknown) => {
            return value;
        });
        return shallow(<Children t={t}/>);
    };

    const getRouterProps = (pathname: string, search?: string) => {
        const location: any = { // tslint:disable-line:no-any
            pathname,
            search
        };
        const routerprops: any = { // tslint:disable-line:no-any
            history: {
                location,
            },
            location,
            match: {
                params: {
                }
            }
        };
        return routerprops;
    };
    it('matches snapshot #/', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot #/devices/', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot #/devices.../device1', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/details/identity/', 'id=device1')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot #/devices.../device1 with interface', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/detail/digitalTwins/properties/', 'id=device1&interfaceId=urn:some:interface:name:1')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot with missing #/devices.../device1', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/details/identity/')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot #/devices.../device1 missing interface', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/detail/digitalTwins/properties', 'id=device1')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot with malformed #/devices.../device1', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/details/identity', 'id=')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot #/devices.../device1 with malformed interface', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/detail/digitalTwins/properties', 'id=device1&interfaceid=')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('matches snapshot #/devices.../device1 with interface and missing deviceId', () => {
        const wrapper = testWithLocalizationContext(
            <Breadcrumb
                hubName="foo.azure-devices.net"
                {...getRouterProps('#/devices/detail/digitalTwins/properties/', 'interfaceId=urn:some:interface:name:1')}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
