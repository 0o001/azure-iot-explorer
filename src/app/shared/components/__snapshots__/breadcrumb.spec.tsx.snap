// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/breadcrumb matches snapshot #/ 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "key": "#",
        "text": undefined,
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot #/devices.../device1 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "href": "#/devices/detail/identity/?id=device1",
        "key": "device_device1",
        "text": "device1",
      },
      Object {
        "key": "identity",
        "text": "deviceContent.navBar.identity",
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot #/devices.../device1 missing interface 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "href": "#/devices/detail/identity/?id=device1",
        "key": "device_device1",
        "text": "device1",
      },
      Object {
        "key": "properties",
        "text": "deviceContent.navBar.properties",
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot #/devices.../device1 with interface 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "href": "#/devices/detail/identity/?id=device1",
        "key": "device_device1",
        "text": "device1",
      },
      Object {
        "href": "#/devices/detail/digitalTwins/interfaces/?id=device1&interfaceId=urn:some:interface:name:1",
        "key": "device_device1_urn:some:interface:name:1",
        "text": "urn:some:interface:name:1",
      },
      Object {
        "key": "properties",
        "text": "deviceContent.navBar.properties",
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot #/devices.../device1 with interface and missing deviceId 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "href": "detail",
        "key": "detail",
        "text": undefined,
      },
      Object {
        "key": "properties",
        "text": "deviceContent.navBar.properties",
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot #/devices.../device1 with malformed interface 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "href": "#/devices/detail/identity/?id=device1",
        "key": "device_device1",
        "text": "device1",
      },
      Object {
        "key": "properties",
        "text": "deviceContent.navBar.properties",
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot #/devices/ 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot with malformed #/devices.../device1 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "href": "details",
        "key": "details",
        "text": undefined,
      },
      Object {
        "key": "identity",
        "text": "deviceContent.navBar.identity",
      },
    ]
  }
/>
`;

exports[`components/shared/breadcrumb matches snapshot with missing #/devices.../device1 1`] = `
<StyledBreadcrumbBase
  className="device-content-breadcrumb"
  items={
    Array [
      Object {
        "key": "Hub",
        "text": "breadcrumb.hub",
      },
      Object {
        "href": "#/devices",
        "key": "Devices",
        "text": "breadcrumb.devices",
      },
      Object {
        "href": "details",
        "key": "details",
        "text": undefined,
      },
      Object {
        "key": "identity",
        "text": "deviceContent.navBar.identity",
      },
    ]
  }
/>
`;
